### PR TITLE
CRIMAP-525 Implement first few document endpoints

### PR DESCRIPTION
**Pull request by zheileman**
## Description of change
This is a first part of a series of several PRs, not only on datastore but also counterpart API gem, to operate with documents in an S3 bucket.

As part of this first PR a little bit of ground preparation has been done and 2 endpoints implemented, to obtain presigned URLs for uploads and downloads.

I will raise a similar PR on the datastore API gem to be able to call these endpoints.

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMAP-525

## Notes for reviewer / how to test
I've marked temporarily as `nocov` a couple of methods that are not exercise in this first PR but will be in second PR.
I've shared on slack a document on how to test these PRs.

### CODE REVIEW

File: spec/support/shared_examples/document_operations.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
```
Comment (timpeat) [inline spec/support/shared_examples/document_operations.rb:1]: shouldn't need this comment ^

**Review by timpeat — APPROVED** [review]
Great start! 🚀

File: spec/support/shared_examples/document_operations.rb (hunk 1 above)
Comment (zheileman) [reply spec/support/shared_examples/document_operations.rb]: True, I've removed it 😄

**Review by willmcb — APPROVED** [review]
👍

**Comment by willmcb** [comment]
Really great, thanks :)

**Pull request merged**